Plugins/Public/playercntl_plugin/PimpShip.cpp

```diff
@@ -271,10 +279,60 @@ namespace PimpShip
 		if (!mapInfo[iClientID].bInPimpDealer || !set_bEnablePimpShip)
 			return false;
 
-		PrintUserCmdText(iClientID, L"Available items: %d", mapAvailableItems.size());
-		for (map<uint, ITEM_INFO>::iterator iter = mapAvailableItems.begin(); iter != mapAvailableItems.end(); iter++)
+		uint beginFrom = 1;
+		uint endAt = mapAvailableItems.size();
+
+		ushort index = 0;
+		for (wstring::const_iterator it = wscParam.begin(); it != wscParam.end(); it++)
+		{
+			index++;
+
+			if (*it == '-')
+			{
+				if (it == wscParam.begin() && it == wscParam.end() - 1)
+				{
+					beginFrom = 1;
+					endAt = mapInfo[iClientID].mapCurrEquip.size();
+				}
+				else if (it == wscParam.end() - 1)
+				{
+					beginFrom = ToUInt(wscParam.substr(0, index));
```

Comment:
There's a lot of these types of calls in this code - what should we do if the user tries to put in non-numeric characters?

Reply:
It will be parsed as 0, doesn't hurt actually.